src/index.ts

```diff
@@ -0,0 +1,25 @@
+import {deriveMasterSK} from "./key-derivation";
+// @ts-ignore
+import secureRandom from "secure-random";
+import {mnemonicToSeedSync, validateMnemonic} from "bip39";
+import assert from "assert";
+import {HDNode} from "./hdnode";
+import {Buffer} from "buffer";
+
+export function generateRandomBLSSecretKey(entropy?: Buffer): Buffer {
```

Comment:
you probably don't need to add "BLS" in the function name here and below

Maybe also worth adding a function comment that explains that `entropy` will be _additional_ entropy, not the entire entropy.